processing/src/main/java/org/apache/druid/segment/join/table/IndexedTableJoinable.java

```diff
@@ -105,7 +106,7 @@ public ColumnValuesWithUniqueFlag getNonNullColumnValues(String columnName, fina
     try (final IndexedTable.Reader reader = table.columnReader(columnPosition)) {
       // Sorted set to encourage "in" filters that result from this method to do dictionary lookups in order.
       // The hopes are that this will improve locality and therefore improve performance.
-      final Set<String> allValues = createValuesSet();
+      final Set<String> allValues = createOrderedValuesSet();
```

Comment:
FYI, InDimFilter wants a sorted set internally, so if you give it a non-sorted set, it'll need to copy it, which slows things down. So we really do want to keep this being a sorted set.

Also, the comment is inaccurate, since InDimFilter always uses sorted sets internally, since #12517. (See ValuesSet.) Could you update it? Something like "Use a SortedSet so InDimFilter doesn't need to create its own".

Reply:
Makes sense, thanks